documentation/articles/conditional_compilation.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Conditional Compilation
+description: Compilation directives for platform specific compilation of code.
+---
+
+# Conditional compilation
+
+MonoGame supports running C# projects on multiple platforms and supports several [directives](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives) to allow developers to selectively include or exclude code from compilation, based on whether certain [scripting symbols](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/symbols) or definitions are defined or not defined.
+
+You can read more broadly about how these directives work in C# in Microsoft’s user documentation on [C# preprocessor directives](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives)
+
+The hash `#` character in front of the **if** and **endif** indicates that these statements are “directives”, and they are handled during the build/compilation process, rather than at runtime.
+
+As an example, you can see in the code excerpt below, one set of code will be compiled on a DesktopGL platform, else the other code will be used.
+
+```csharp
+#if DESKTOPGL
+    // The Client is running on the DesktopGL runtime
+#else
+    // The Client is NOT running on DesktopGL
+#endif
+```
+
+The critical factor is that the above pattern reduces what code is compiled and is included in the final build / release.
+
+> [!NOTE]
+> The above technique is an alternative to using the `MonoGamePlatform` method, where all code included is compiled and included in your build, for example:
+>
+> ```csharp
+> if(PlatformInfo.MonoGamePlatform == MonoGamePlatform.DesktopGL)
+> {
+>     // The Client is running on the DesktopGL runtime

```

Comment:
Should we note that note all api's are available across all platforms? When using this Runtime method you cannot use Android or Steam API's for example, since the assembly won't be available.

For platform specific code (i.e Play Store integration etc) you really need to use #defines or include specific files in the head project.

Reply:
Yeah, it is the one limitation with this only runtime method.  Ideally, MG should expose some defines that can be consumed similar to other engines, but without an actual runtime component, I'm not sure it is possible.

Reply:
afaik, each platform Nuget has a build Targets file. The `DefineConstants` MSbuild property could be added to if they are not already like

```
<DefineConstants>$(DefineConstants);__MONOGAME_ANDROID__<DefineConstants>
```

or 

```
<DefineConstants>$(DefineConstants);__MONOGAME_DESKTOPGL__<DefineConstants>
```

if that is what you are thinking of?

See https://github.com/MonoGame/MonoGame/blob/249dc0dfa3d8c2182213fa03b4357d1e063cd71e/MonoGame.Framework/MonoGame.Framework.DesktopGL.targets#L4

and https://github.com/MonoGame/MonoGame/blob/249dc0dfa3d8c2182213fa03b4357d1e063cd71e/MonoGame.Framework/MonoGame.Framework.Android.targets

Reply:
@harry-cpp said something similar but the conversation didn't progress beyond that.

If those features could give us compile-time constants for MonoGame Platforms, then yes, that would be the answer (along the style naming defined in the guide)